flask_mongorest/resources.py

```diff
@@ -199,15 +201,26 @@ def serialize_field(self, obj, **kwargs):
         else:
             return self.serialize(obj, **kwargs)
 
+    def _subresource(self, obj):
+        """ Selects and creates an appropriate sub-resource class for delegation or return None if there isn't one """
+        default_s_class = self._child_document_resources[self.default_child_resource_document] if self.default_child_resource_document else None
+        s_class = self._child_document_resources.get(obj.__class__, default_s_class)
+        if s_class and s_class != self.__class__:
+            r = s_class()
+            r.data = self.data
+            return r
+        else:
+            return None
```

Comment:
Return value of `None` indicates that there is no subclass to delegate to (think simple, non-inherited resources). Existence of the result is a signal to caller to perform actual delegation.